app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -4232,10 +4268,38 @@ class ChatActivity :
             CONTENT_TYPE_LINK_PREVIEW -> message.isLinkPreview()
             CONTENT_TYPE_SYSTEM_MESSAGE -> !TextUtils.isEmpty(message.systemMessage)
             CONTENT_TYPE_UNREAD_NOTICE_MESSAGE -> message.id == "-1"
+            CONTENT_TYPE_CALL_STARTED -> message.id == "-2"
+
             else -> false
         }
     }
 
+    private fun processMostRecentMessage(recent: ChatMessage, chatMessageList: List<ChatMessage>) {
+        when (recent.systemMessageType) {
+            ChatMessage.SystemMessageType.CALL_STARTED -> {
+                if (!callStarted) {
+                    val unreadChatMessage = ChatMessage()
+                    unreadChatMessage.jsonMessageId = CALL_STARTED_ID
+                    unreadChatMessage.actorId = "-2"
+                    val name = if (recent.actorDisplayName.isNullOrEmpty()) "Guest" else recent.actorDisplayName
+                    unreadChatMessage.actorDisplayName = name
+                    unreadChatMessage.actorType = recent.actorType
+                    unreadChatMessage.timestamp = chatMessageList[0].timestamp
+                    unreadChatMessage.message = null
+                    adapter?.addToStart(unreadChatMessage, false)
+                    callStarted = true
+                }
+            } // add CallStartedMessage with id -2
+            ChatMessage.SystemMessageType.CALL_ENDED,
+            ChatMessage.SystemMessageType.CALL_MISSED,
```

Comment:
ChatMessage.SystemMessageType.CALL_TRIED is missing?

Reply:
I left it out intentionally, because the docs had this
- `call_tried` - You tried to call {user}

which, I think, implies that the current user started a call and therefore wouldn't need a CallStartedMessage to be rendered in the first place.

Reply:
start a call from mobile
do not accept on web
hang up again on mobile

--> the CallStartedMessage remains visible but the call is over.
Including ChatMessage.SystemMessageType.CALL_TRIED would fix this.